src/analysis/functions.rs

```diff
@@ -26,21 +26,19 @@ pub fn analyze(env: &Env, type_: &library::Class, class_tid: library::TypeId) ->
 }
 
 fn analyze_function(env: &Env, type_: &library::Function, class_tid: library::TypeId) -> Info {
-    let klass = env.type_(class_tid);
-
     let mut commented = false;
 
     let ret = if type_.ret.typ == Default::default() { None } else { Some(type_.ret.clone()) };
 
     if let Some(ref ret_) = ret {
-        let rust_type = env.type_(ret_.typ).to_parameter_rust_type(ret_.direction);
+        let rust_type = (&env.library, ret_.typ).to_parameter_rust_type(ret_.direction);
```

Comment:
I'm a little worried that we might be fooling ourselves here and this is no different than the conservative
```rust
let rust_type = rust_type::to_parameter(&env.library, ret_.typ, ret_.direction);
```

So the idea of implementing traits for tuples and even giving those tuples names may not have been a good advice after all.